src/QsCompiler/DataStructures/DataTypes.fs

```diff
@@ -83,12 +83,33 @@ type Position = private Position of int * int with
         let column = if b.Line = 0 then a.Column + b.Column else b.Column
         Position (line, column)
 
+    /// <summary>
+    /// Translates the first position backwards by the amount of the second position. If the resulting position is on
+    /// the same line, the column numbers are subtracted; otherwise, the first position's column number is used.
+    /// </summary>
+    /// <exception cref="ArgumentOutOfRangeException">
+    /// Thrown if the resulting position has a negative line or column number.
+    /// </exception>
+    static member (-) (a : Position, b : Position) =
```

Comment:
Similar to what you did with the `(+)` operator, can you provide an example for why this logic is what it is?

Reply:
It's the inverse of `+`, so if `absolute = offset + relative`, then `absolute - offset = relative`. I think the common rule is that the column number of position that occurs later is the one that's preserved. In `offset + relative`, `relative` occurs after `offset`, so `+` keeps the RHS column number. In `absolute - offset`, `absolute` occurs after `offset`, so `-` keeps the LHS column number.

Reply:
Ah okay so the addition operator is not commutative. I didn't see that before. It could be good to put that in the doc for the `(+)`. It could also be good to give more descriptive names to the inputs like what you did in the example you just gave.

Reply:
Yes, sounds good, I updated docs and parameter names for `+` and `-`.